src/NLog/LayoutRenderers/Log4JXmlEventLayoutRenderer.cs

```diff
@@ -94,7 +94,7 @@ public Log4JXmlEventLayoutRenderer(IAppDomain appDomain)
                 CultureInfo.InvariantCulture,
                 "{0}({1})",
                 appDomain.FriendlyName,
-                ProcessIDHelper.Instance.CurrentProcessID);
+                LogFactory.DefaultAppEnvironment.CurrentProcessId);
```

Comment:
Would also expect CurrentAppEnvironment here

Reply:
Not possible to inject AppEnvironment into constructor without it becoming a breaking API change. Planning to make the constructor internal in NLog 5, and change parameter from AppDomain to AppEnvironment. But not possible with NLog 4.7.